ompi/mca/osc/ucx/osc_ucx_comm.c

```diff
@@ -350,23 +350,27 @@ static inline int get_dynamic_win_info(uint64_t remote_addr, ompi_osc_ucx_module
     }
 
     memcpy(&win_count, temp_buf, sizeof(uint64_t));
-    assert(win_count > 0 && win_count <= OMPI_OSC_UCX_ATTACH_MAX);
+    if (win_count > OMPI_OSC_UCX_ATTACH_MAX) {
```

Comment:
@devreal thanks for your detailed look - I think we can inform the user if win_count exceeds ATTACH_MAX, although I'm not sure how this could happen, these asserts may be have been for debugging purposes only, since as @hjelmn pointed out they will not be part of an optimized build anyways.

Reply:
Right, this indeed seems to be something that is caught during registration, not during access. If the assert does not hold it indicates some internal inconsistencies...